promotion/controllers_test.go

```diff
@@ -1292,8 +1293,7 @@ func (suite *ControllersTestSuite) TestSuggestionDrain() {
 	err = s.Sign(privKey, crypto.Hash(0), req)
 	suite.Require().NoError(err)
 
-	anonymousAddress := uuid.Must(uuid.FromString(w.ProviderID))
-	info.AnonymousAddress = &anonymousAddress
+	info.UserDepositDestination = w.ProviderID
```

Comment:
This code seems like the same as before, do we still want to validate the `uuid.Must` property of the ProviderID?

Will this parameter change with the client change? It looks like we're mapping the existing anonymous addresses to `UserDepositDestination`?

Reply:
UserDepositDestination is a string.  Current use cases happen to use a uuid, but that might not always be the case for wallet deposit providers.  In this test we are just filling in a value to make sure the value is right.

We are not directly mapping from anonymous address to destination, they are two seperate values.  The anonymous address is a uuid that the browser creates and we give uphold to obfuscate the actual destination.   Destination is the actual destination for the deposit.

I agree that this test is a bit confusing, it should be more clear that anonymous address and this deposit destination are different things.